nest/test/AggregateTokenCredbull.t.sol

```diff
@@ -0,0 +1,206 @@
+    }
+
+    function testComponentTokenFlow() public {
+        uint256 privateKey = vm.envUint("PRIVATE_KEY");
+        address controller = vm.addr(privateKey);
+        vm.startBroadcast(privateKey);
+
+        // Step 1: Deposit USDC to get aggregate tokens
+        uint256 shares = token.deposit(TEST_AMOUNT, controller, controller);
+        console.log("Deposited shares:", shares);
+
+        // Step 2: Buy component token (Credbull Vault tokens)
+        token.buyComponentToken(CREDBULL_VAULT, TEST_AMOUNT);
+        console.log("Bought Credbull tokens");
+
+        uint256 currentPeriod = CREDBULL_VAULT.currentPeriod();
+        console.log("Current Period:", currentPeriod);
+
+        // Get balances
+        uint256 balance = CREDBULL_VAULT.balanceOf(address(token), currentPeriod);
+        console.log("Balance at current period:", balance);
+
+        // Step 3: We need to make our request through the AggregateToken
+        vm.stopBroadcast();
+        vm.startPrank(address(token)); // Act as the AggregateToken
+
+        // Request redeem as the AggregateToken
+        CREDBULL_VAULT.requestRedeem(
+            TEST_AMOUNT,
+            address(token), // controller
+            address(token) // owner
+        );
+        console.log("Redeem requested");
+
+        // Step 4: Wait notice period
+        uint256 noticePeriod = CREDBULL_VAULT.noticePeriod();
+        vm.warp(block.timestamp + noticePeriod * 1 days);
+        console.log("Time warped");
+
+        vm.stopPrank();
+        vm.startBroadcast(privateKey);
+
+        // Step 5: Now try to sell component token
+        token.sellComponentToken(CREDBULL_VAULT, TEST_AMOUNT);
+        console.log("Successfully sold Credbull tokens");
+
+        vm.stopBroadcast();
+    }
+
+    function testQueryBalances() public view {
+        console.log("\n=== Current Balances ===");
+        console.log("USDC Balance:", USDC.balanceOf(account));
+        console.log("USDC Allowance (AggregateToken):", USDC.allowance(account, address(token)));
+        console.log("USDC Allowance (Credbull):", USDC.allowance(account, address(CREDBULL_VAULT)));
+        console.log("Aggregate Token Balance:", token.balanceOf(account));
+
+        uint256 currentPeriod = CREDBULL_VAULT.currentPeriod();
+        console.log("\n=== Credbull Vault Info ===");
+        console.log("Current Period:", currentPeriod);
+        console.log("Notice Period:", CREDBULL_VAULT.noticePeriod());
+        console.log("Asset:", CREDBULL_VAULT.asset());
+        console.log("Credbull Balance:", CREDBULL_VAULT.balanceOf(address(token), currentPeriod));
+        console.log("AggregateToken Balance in Vault:", CREDBULL_VAULT.balanceOf(address(token), currentPeriod));
+        console.log("Unlock Request Amount:", CREDBULL_VAULT.unlockRequestAmount(address(token), currentPeriod));
+        console.log("Claimable Request Amount:", CREDBULL_VAULT.claimableRedeemRequest(currentPeriod, address(token)));
+    }
```

Comment:
Maybe make this into a `function printQueryBalances()` and if you need to print it out do:

```
function testQueryBalances() public view {
  printQueryBalances();
}
```

Reply:
done